log/Logger.js

```diff
@@ -1,26 +1,48 @@
-const moment = require('moment')
-const winston = require('winston')
+const util = require('util');
+const {createLogger, format, transports} = require('winston');
+const {errors, json, metadata, timestamp} = format;
 
-const container = new winston.Container()
-
-function timestamp() {
-  return moment().format('YYYY-MM-DDTHH:mm:ssZ')
-}
-
-function transport(label) {
-  return new winston.transports.Console({
-    level: (process.env.LOG_LEVEL || 'INFO').toLowerCase(),
-    json: process.env.JSON_PRINT || false,
-    timestamp: timestamp,
-    label: label
-  })
-}
+const customMessageFormat = format((info) => {
+  const {message, metadata} = info;
+  if (metadata && typeof metadata === 'object' && Object.keys(metadata).length) {
+    info.message = message + util.inspect(metadata);
+  }
+  // delete metadata as its concatenated to the message
+  delete info.metadata;
 
-class Logger {
+  return info;
+});
 
-  static getLogger(name) {
-    return container.add(name, {transports: [transport(name)]})
-  }
-}
+/**
+ * This logger outputs the logs in json format with fields
+ *   level: log level,
+ *   timestamp: timestamp the message was received,
+ *   moduleName: name of the module/file logging the message,
+ *   message: logger statement with dynamic data,
+ *   stack: error stack trace if any
+ * e.g.
+ * {
+ *   "level":"info",
+ *   "timestamp":"2021-01-14T11:40:54.859Z",
+ *   "message":"Hello World { hello: 'world' }",
+ *   "moduleName":"loggerTest",
+ *   "stack": "Error: ...."
+ * }
+ */
+const logger = createLogger({
+  level: process.env.LOG_LEVEL || 'info',
+  format: format.combine(
+    timestamp(),
+    metadata({fillExcept: ['level', 'timestamp', 'message', 'moduleName', 'stack']}),
+    customMessageFormat(),
+    errors({stack: true}),
+    json()
+  ),
+  transports: [
+    new transports.Console({
+      level: process.env.LOG_LEVEL || 'verbose'
```

Comment:
```suggestion
      level: process.env.LOG_LEVEL || 'info'
```
To avoid overly verbose logs in environments which do not explicitly set a log level

Reply:
inadvertent change, reverted